algebird-core/src/main/scala/com/twitter/algebird/QTree.scala

```diff
@@ -63,8 +61,26 @@ object QTree {
   def apply(k: Double): QTree[Double] = apply(k -> k)
 }
 
-class QTreeSemigroup[A: Monoid](k: Int) extends Semigroup[QTree[A]] {
+class QTreeSemigroup[A](k: Int)(implicit val underlyingMonoid: Monoid[A]) extends Semigroup[QTree[A]] {
```

Comment:
they are totally equivalent. If we want a blue bike shed I could say:

```scala
class QTreeSemigroup[A: Monoid](k: bits) extends Semigroup[QTree[A]] {
  def underlyingMonoid = implicitly[Monoid[A]]
  ...
}
```
that was just more characters. @jnievelt even in the type-bound case you can pass it explicitly if you want. So
```scala
def method[T: TypeClass1: TypeClass2](implicit tc: TypeClass3[T]) = ...
```
will be desugared to:
```scala
def method[T](implicit <somecompilergenid>: TypeClass1[T], <somecompilergenid>: TypeClass2[T], tc: TypeClass3[T]) = ...
```